File: AlphaStream/Models/Author.py
from datetime import datetime

class Author(object):
    """ Author object for storing user information """

    def __init__(self, json):

        # Unique author ID hash
        self.Id = json['id']

        # List of alphas IDs by the author
        self.Alphas = json.get('alphas', [])

        # Number of author's alphas listed on the market
        self.AlphasListed = json.get('alphas-listed', None)

        # Average number of hours the author spent coding on an alpha
        self.AnalysisAverageLength = json.get('analysis-average-length', None)

        # Total number of backtests performed by the author
        self.Backtests = json.get('backtests', None)

        # Biography provided by the author
        self.Biography = json.get('biography', None)

        # Number of forum discussions the author has participated in
        self.ForumDiscussions = json.get('forum-discussions', None)

        # Number of comments the author has posted in the community forum
        self.ForumComments = json.get('forum-comments', None)

        # Author's preferred coding language
        self.Language = json.get('language', None)

        # UTC time the author was last online
        self.LastOnlineTime = datetime.utcfromtimestamp(json['last-online-time']) if 'last-online-time' in json else None

        # Author's location - city, state, country
        self.Location = json.get('location', None)

        # Total number of projects the author has created
        self.Projects = json.get('projects', None)

        # UTC time the author signed up with QuantConnect
        self.SignUpTime = datetime.utcfromtimestamp(json['signup-time']) if 'signup-time' in json else None

        # Author's social media profile
        self.SocialMedia = json.get('social-media', None)

    def __repr__(self):
        return f'Alpha Stream Author {self.Id[:5]:>5} is from {self.Location}. Signed up {self.SignUpTime} and codes in {self.Language}'

File: QuantConnect.AlphaStream/Models/Author.cs
using Newtonsoft.Json;
using QuantConnect.Util;
using System;
using System.Collections.Generic;
using System.Text;

namespace QuantConnect.AlphaStream.Models
{
    /// <summary>
    /// Author user of QuantConnect responsible for creation of an Alpha.
    /// </summary>
    public class Author
    {
        /// <summary>
        /// Unique string hash id for Author.
        /// </summary>
        [JsonProperty("id")]
        public string Id { get; set; }

        /// <summary>
        /// String array of Alpha-Id hashes the Author has created.
        /// </summary>
        [JsonProperty("alphas")]
        public List<string> Alphas { get; set; } = new List<string>();

        /// <summary>
        /// Number of Alphas Author has listed in the marketplace.
        /// </summary>
        [JsonProperty("alphas-listed")]
        public int AlphasListed { get; set; }

        /// <summary>
        /// Number of days of the average analysis(backtest) length for the Author.
        /// </summary>
        [JsonProperty("analysis-average-length")]
        public int? AnalysisAverageLength { get; set; }

        /// <summary>
        /// Lifetime number of analysis(backtests) conducted by the Author.
        /// </summary>
        [JsonProperty("backtests")]
        public int Backtests { get; set; }

        /// <summary>
        /// Profile auto-biography (100-500 words).
        /// </summary>
        [JsonProperty("biography")]
        public string Biography { get; set; }

        /// <summary>
        /// Number of forum discussions started by the Author.
        /// </summary>
        [JsonProperty("forum-discussions")]
        public int ForumDiscussions { get; set; }

        /// <summary>
        /// Number of forum comments made by the Author.
        /// </summary>
        [JsonProperty("forum-comments")]
        public int ForumComments { get; set; }

        /// <summary>
        /// Researcher preferred programming language.
        /// </summary>
        [JsonProperty("language")]
        public Language Language { get; set; }

        /// <summary>
        /// Unix timestamp of the last time the Author was online (Updated daily).
        /// </summary>
        [JsonProperty("last-online-time"), JsonConverter(typeof(DoubleUnixSecondsDateTimeJsonConverter))]
        public DateTime? LastOnlineTime { get; set; }

        /// <summary>
        /// Best known estimate of the Author geographic location.
        /// </summary>
        [JsonProperty("location")]
        public string Location { get; set; }

        /// <summary>
        /// Total count of the number of projects the Author has in account.
        /// </summary>
        [JsonProperty("projects")]
        public int Projects { get; set; }

        /// <summary>
        /// Unix timestamp of the user sign up.
        /// </summary>
        [JsonProperty("signup-time"), JsonConverter(typeof(DoubleUnixSecondsDateTimeJsonConverter))]
        public DateTime SignupTime { get; set; }

        /// <summary>
        /// Array of social media profile links.
        /// </summary>
        [JsonProperty("social-media")]
        public List<string> SocialMedia { get; set; } = new List<string>();

        /// <summary>
        /// Returns a string that represents the Author object
        /// </summary>
        /// <param name="extended">False if we want the short version</param>
        /// <returns>A string that represents the Author object</returns>
        public string ToString(bool extended)
        {
            var stringBuilder = new StringBuilder($"Author Id:\t{Id}");

            if (Alphas.Count > 0)
            {
                stringBuilder.Append($"{Environment.NewLine}Alphas listed:\t{AlphasListed}");
                stringBuilder.Append($"{Environment.NewLine}Alphas:\t{string.Join(", ", Alphas)}");
            }

            stringBuilder.Append($"{Environment.NewLine}Location:\t{Location}");
            stringBuilder.Append($"{Environment.NewLine}Language:\t{Language}");
            stringBuilder.Append($"{Environment.NewLine}Projects:\t{Projects}");
            stringBuilder.Append($"{Environment.NewLine}Backtests:\t{Backtests}");
            stringBuilder.Append($"{Environment.NewLine}Analysis average length:\t{AnalysisAverageLength}");

            if (!extended)
            {
                return stringBuilder.ToString();
            }

            stringBuilder.Append($"{Environment.NewLine}Biography:\t{Biography.Substring(0, 100)}...");
            stringBuilder.Append($"{Environment.NewLine}Sign-up time:\t{SignupTime}");

            if (LastOnlineTime.HasValue)
            {
                stringBuilder.Append($"{Environment.NewLine}Last time online:\t{LastOnlineTime}");
            }

            if (ForumDiscussions > 0)
            {
                stringBuilder.Append($"{Environment.NewLine}Forum discussions:\t{ForumDiscussions}");
            }

            if (ForumComments > 0)
            {
                stringBuilder.Append($"{Environment.NewLine}Forum comments:\t{ForumComments}");
            }

            return stringBuilder.ToString();
        }

        /// <summary>
        /// Returns a string that represents the Author object
        /// </summary>
        /// <returns>A string that represents the Author object</returns>
        public override string ToString() => ToString(true);

        /// <summary>
        /// Returns a string that represents the Author object
        /// </summary>
        /// <param name="extended">False if we want the short version</param>
        /// <returns>A string that represents the Author object</returns>
        public string ToStringInline(bool extended = false) => ToString(extended).Replace(Environment.NewLine, " ");
    }
}